diff --git a/testing/mochitest/tests/SimpleTest/EventUtils.js b/testing/mochitest/tests/SimpleTest/EventUtils.js
index f5cdcc756f74ebed56f108f24961d0401ef3ba50..4c87efa57aa4d23a1ab6cf7cc94de851fcafb8a1 100644
--- a/testing/mochitest/tests/SimpleTest/EventUtils.js
+++ b/testing/mochitest/tests/SimpleTest/EventUtils.js
@@ -1734,22 +1734,26 @@ function synthesizeAndWaitNativeMouseMove(
  * @param {number} [aEvent.repeat]
  *        If you emulate auto-repeat, you should set the count of repeat.
  *        This method will automatically synthesize keydown (and keypress).
- * @param {*} aEvent.location
+ * @param {boolean} [aEvent.ctrlKey]
+ *        Customise the `ctrlKey` modifier.
+ * @param {boolean} [aEvent.shiftKey]
+ *        Customise the `shiftKey` modifier.
+ * @param {*} [aEvent.location]
  *        If you want to specify this, you can specify this explicitly.
  *        However, if you don't specify this value, it will be computed
  *        from code value.
- * @param {string} aEvent.type
+ * @param {string} [aEvent.type]
  *        Basically, you shouldn't specify this.  Then, this function will
  *        synthesize keydown (, keypress) and keyup.
  *        If keydown is specified, this only fires keydown (and keypress if
  *        it should be fired).
  *        If keyup is specified, this only fires keyup.
- * @param {number} aEvent.keyCode
+ * @param {number} [aEvent.keyCode]
  *        Must be 0 - 255 (0xFF). If this is specified explicitly,
  *        .keyCode value is initialized with this value.
  * @param {DOMWindow} [aWindow=window]
  *        DOM window used to dispatch the event.
- * @param {Function} aCallback
+ * @param {Function} [aCallback]
  *        Is optional and can be used to receive notifications from TIP.
  *
  * @description
@@ -3637,7 +3641,7 @@ function _computeSrcElementFromSrcSelection(aSrcSelection) {
  *                The final x coordinate inside srcElement
  * @param {number} aParams.finalY
  *                The final x coordinate inside srcElement
- * @param {Any} aParams.id
+ * @param {any} aParams.id
  *                The pointer event id
  * @param {DOMWindow} aParams.srcWindow
  *                The DOM window for dispatching event on srcElement, defaults to the current window object.
