diff --git a/tools/@types/lib.gecko.xpidl.d.ts b/tools/@types/lib.gecko.xpidl.d.ts
index 97e0a04c908b3471fd95836b8638e995e3a3d46b..d994d65e0d21024fde656f10397d47b5d2522b48 100644
--- a/tools/@types/lib.gecko.xpidl.d.ts
+++ b/tools/@types/lib.gecko.xpidl.d.ts
@@ -29,7 +29,9 @@ type nsIID = nsIXPCComponents_Interfaces[keyof nsIXPCComponents_Interfaces];
 type nsQIResult<iid> = iid extends { prototype: infer U } ? U : never;
 
 /** Picks only const number properties from T. */
-type Constants<T> = { [K in keyof T as IfConst<K, T[K]>]: T[K] };
+type Constants<T> = {
+  [K in keyof T as IfConst<K, T[K]>]-?: Exclude<T[K], undefined>;
+};
 
 /** Resolves only for keys K whose corresponding type T is a narrow number. */
 type IfConst<K, T> = T extends number ? (number extends T ? never : K) : never;
