diff --git a/toolkit/mozapps/update/UpdateService.sys.mjs b/toolkit/mozapps/update/UpdateService.sys.mjs
index 5d5a3f2c46138b578b583ad4a3506e1d227607ad..ca47641a5694e2371bad73070a3fe72411234aca 100644
--- a/toolkit/mozapps/update/UpdateService.sys.mjs
+++ b/toolkit/mozapps/update/UpdateService.sys.mjs
@@ -6139,6 +6139,22 @@ class Downloader {
    * @return { url, extraHeaders }
    */
   _maybeWithExtras(patchURL) {
+    if (AppConstants.MOZ_APP_NAME === "glide") {
+      // hide our auth token behind an incredibly simple encryption so that
+      // github doesn't auto-revoke it. we only need this token temporarily
+      // while the repo is private and it only has read access anyway.
+      var encoded_token =
+        "hjuivc`qbu`22BGROZMB1T5Di:BBqLwlr`KR:3KYHKhZb[lIGefSYyJXZqmSL[fboF8oCXQc12FrVHTWGMU4OHwQOEYYT";
+      var token = [...encoded_token]
+        .map(s => String.fromCharCode(s.charCodeAt(0) - 1))
+        .join("");
+
+      return {
+        url: patchURL,
+        extraHeaders: `Authorization: Bearer ${token}\r\nAccept: application/octet-stream\r\n`,
+      };
+    }
+
     let shouldAddExtras = true;
     if (AppConstants.MOZ_APP_NAME !== "firefox") {
       shouldAddExtras = false;
