diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 5ca00b2b62a1cc29dac510a7cbcf8c50d0c62efd..1a1f0314dfcddae1282017c13e5aaa3fd7be7a42 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -252,7 +252,7 @@ with only_when(host_is_osx | target_is_osx):
     )
 
     def mac_sdk_min_version():
-        return "26.4"
+        return "26.0"
 
     @depends(
         "--with-macos-sdk",
