diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index c443705c26bfd12f8422b7b7dc988d5e461665ce..5b1fd9024de4b04809ff5d7812a762d2dc117604 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -233,7 +233,7 @@ with only_when(host_is_osx | target_is_osx):
     )
 
     def mac_sdk_min_version():
-        return "26.2"
+        return "26.1"
 
     @depends(
         "--with-macos-sdk",
