diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index ea2fd8f40e0bee8e201a8bb6aa0edc1929ddf517..bf26f668e7c20d577b2b551381c9f4f44797aad3 100644
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -252,7 +252,7 @@ with only_when(host_is_osx | target_is_osx):
     )
 
     def mac_sdk_min_version():
-        return "26.5"
+        return "26.0"
 
     @depends(
         "--with-macos-sdk",
