diff --git a/testing/mochitest/browser-test.js b/testing/mochitest/browser-test.js
index a687d271a91207f3e4a7b9afdf67031f8d9385d1..07492dd930569d82b55ee3ee5941c63240fc7f5b 100644
--- a/testing/mochitest/browser-test.js
+++ b/testing/mochitest/browser-test.js
@@ -243,6 +243,16 @@ function Tester(aTests, structuredLogger, aCallback) {
   this.PerTestCoverageUtils = ChromeUtils.importESModule(
     "resource://testing-common/PerTestCoverageUtils.sys.mjs"
   ).PerTestCoverageUtils;
+  // glide-injection-start
+  this.GlideTestUtils = ChromeUtils.importESModule(
+    "resource://testing-common/GlideTestUtils.sys.mjs",
+    { global: "current" }
+  ).GlideTestUtils;
+  this.GlideEventUtils = ChromeUtils.importESModule(
+    "chrome://glide/content/event-utils.mjs",
+    { global: "current" }
+  );
+  // glide-injection-end
 
   this.PromiseTestUtils.init();
 
@@ -1411,6 +1421,31 @@ Tester.prototype = {
     scope.BrowserTestUtils = this.BrowserTestUtils;
     scope.TestUtils = this.TestUtils;
     scope.ExtensionTestUtils = this.ExtensionTestUtils;
+
+    // glide-injection-start
+    Object.defineProperty(scope, "glide", {
+      get() {
+        return GlideBrowser.api;
+      },
+    });
+    scope.GlideEventUtils = this.GlideEventUtils;
+    scope.keys = scope.GlideEventUtils.synthesize_keyseq;
+    scope.until = this.GlideTestUtils.until;
+    scope.waiter = this.GlideTestUtils.waiter;
+    scope.GlideTestUtils = this.GlideTestUtils;
+    scope.wait_for_mode = this.GlideTestUtils.wait_for_mode.bind(
+      this.GlideTestUtils
+    );
+    scope.reload_config = this.GlideTestUtils.reload_config.bind(
+      this.GlideTestUtils
+    );
+    scope.write_config = this.GlideTestUtils.write_config.bind(
+      this.GlideTestUtils
+    );
+    this.GlideTestUtils.__load_globals(scope);
+    this.GlideTestUtils.__load_globals({ gBrowser: window.gBrowser });
+    // glide-injection-end
+
     // Pass a custom report function for mochitest style reporting.
     scope.Assert = new this.Assert(function (err, message, stack) {
       currentTest.addResult(
@@ -1759,6 +1794,16 @@ function testScope(aTester, aTest, expected) {
       self.record(condition, name);
     }
   };
+  // glide-injection-start
+  this.notok = function test_notok(condition, name) {
+    if (arguments.length > 2) {
+      const ex = "Too many arguments passed to notok(condition, name)`.";
+      self.record(false, name, ex);
+    } else {
+      self.record(!condition, name);
+    }
+  };
+  // glide-injection-end
   this.record = function test_record(condition, name, ex, stack, expected) {
     if (expected == "fail") {
       aTest.addResult(
@@ -1792,6 +1837,18 @@ function testScope(aTester, aTest, expected) {
       Components.stack.caller
     );
   };
+  // glide-injection-start
+  this.isjson = function test_is(a, b, name) {
+    self.is(
+      JSON.stringify(a, typeof a === "object" ? Object.keys(a).sort() : null),
+      JSON.stringify(b, typeof b === "object" ? Object.keys(b).sort() : null),
+      name
+    );
+  };
+  // this.keys = async keyseq => {
+  //   await this.GlideTestUtils(keyseq);
+  // };
+  // glide-injection-start
   this.isfuzzy = function test_isfuzzy(a, b, epsilon, name) {
     self.record(
       a >= b - epsilon && a <= b + epsilon,
@@ -1970,6 +2027,17 @@ function testScope(aTester, aTest, expected) {
     }
   };
 
+  // glide-injection-start
+  this.sleep_frames = async function sleep_frames(count) {
+    for (let i = 0; i < count; i++) {
+      await new Promise(r => requestAnimationFrame(r));
+    }
+  };
+  this.sleep_forever = async function sleep_frames() {
+    await this.sleep_frames(999999999999999);
+  };
+  // glide-injection-end
+
   this.requestCompleteLog = function test_requestCompleteLog() {
     self.__tester.structuredLogger.deactivateBuffering();
     self.registerCleanupFunction(function () {
@@ -2008,6 +2076,10 @@ testScope.prototype = {
   TestUtils: null,
   ExtensionTestUtils: null,
   Assert: null,
+  // glide-injection-start
+  GlideTestUtils: null,
+  GlideEventUtils: null,
+  // glide-injection-end
 
   /**
    * Add a function which returns a promise (usually an async function)
