diff --git a/testing/mochitest/runtests.py b/testing/mochitest/runtests.py
index 522906776da01f69bc2f873db0159e0b0dce81f9..9c566be262801943713d3cfa69d36950d5dd0848 100644
--- a/testing/mochitest/runtests.py
+++ b/testing/mochitest/runtests.py
@@ -2360,6 +2360,15 @@ toolbar#nav-bar {
             path = os.path.join(profile_data_dir, profile)
             self.profile.merge(path, interpolation=interpolation)
 
+            # glide-injection-start
+            glide_dir = os.path.join(path, "glide")
+            if os.path.isdir(glide_dir):
+              dest = os.path.join(options.profilePath, "glide")
+              if os.path.exists(dest):
+                  shutil.rmtree(dest)
+              shutil.copytree(glide_dir, dest)
+            # glide-injection-end
+
     @property
     def conditioned_profile_copy(self):
         """Returns a copy of the original conditioned profile that was created."""
@@ -2724,6 +2733,12 @@ toolbar#nav-bar {
             attrs = ["pid", "ppid", "name", "cmdline", "username"]
             for proc in psutil.process_iter():
                 try:
+                    if "firefoxpwa" in proc.name():
+                        # no point in reporting a warning for firefoxpwa instances
+                        # as they should only be running in local testing, and are
+                        # almost guaranteed to not be running a marionette server.
+                        continue
+
                     if "firefox" in proc.name():
                         firefoxes = f"{firefoxes}{proc.as_dict(attrs=attrs)}\n"
                 except Exception:
