diff --git a/browser/components/DesktopActorRegistry.sys.mjs b/browser/components/DesktopActorRegistry.sys.mjs
index 4ceb32d5517d39074842e78c4f48b06c48e0881d..0665219ab80a42ea8211d15c0d3484dc43619c75 100644
--- a/browser/components/DesktopActorRegistry.sys.mjs
+++ b/browser/components/DesktopActorRegistry.sys.mjs
@@ -43,6 +43,15 @@ let JSPROCESSACTORS = {
     remoteTypes: ["parent", "privilegedabout"],
   },
 
+  GeminiProtocol: {
+    parent: {
+      esModuleURI: "resource:///actors/GeminiProtocolParent.sys.mjs",
+    },
+    includeParent: true,
+    // gemini pages are loaded in regular web content processes
+    safeForUntrustedWebProcess: true,
+  },
+
   RefreshBlockerObserver: {
     child: {
       esModuleURI: "resource:///actors/RefreshBlockerChild.sys.mjs",
@@ -96,6 +105,54 @@ let JSWINDOWACTORS = {
     remoteTypes: ["parent"],
   },
 
+  GlideDocs: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideDocsParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideDocsChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+      },
+    },
+    matches: ["resource://glide-docs/dynamic/mappings.html"],
+    allFrames: false,
+    includeChrome: false,
+  },
+  GlideTutor: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideTutorParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideTutorChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+        click: {},
+      },
+    },
+    matches: ["resource://glide-tutor/*"],
+    allFrames: false,
+    includeChrome: false,
+  },
+
+  GlideHandler: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideHandlerParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideHandlerChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+        focusin: {},
+        blur: {},
+        keydown: {},
+      },
+    },
+    allFrames: true,
+    includeChrome: true,
+    safeForUntrustedWebProcess: true,
+  },
+
   AboutLogins: {
     parent: {
       esModuleURI:
