diff --git a/browser/components/DesktopActorRegistry.sys.mjs b/browser/components/DesktopActorRegistry.sys.mjs
index 059edd72c3b220ee3ec2fa4becf1fa1b67627834..1e3f7bfc1becef5a604a5133098ce00982b40aaa 100644
--- a/browser/components/DesktopActorRegistry.sys.mjs
+++ b/browser/components/DesktopActorRegistry.sys.mjs
@@ -41,6 +41,13 @@ let JSPROCESSACTORS = {
     includeParent: true,
   },
 
+  GeminiProtocol: {
+    parent: {
+      esModuleURI: "resource:///actors/GeminiProtocolParent.sys.mjs",
+    },
+    includeParent: true,
+  },
+
   RefreshBlockerObserver: {
     child: {
       esModuleURI: "resource:///actors/RefreshBlockerChild.sys.mjs",
@@ -92,6 +99,53 @@ let JSWINDOWACTORS = {
     enablePreference: "browser.contextual-password-manager.enabled",
   },
 
+  GlideDocs: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideDocsParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideDocsChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+      },
+    },
+    matches: ["resource://glide-docs/dynamic/mappings.html"],
+    allFrames: false,
+    includeChrome: false,
+  },
+  GlideTutor: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideTutorParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideTutorChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+        click: {},
+      },
+    },
+    matches: ["resource://glide-tutor/*"],
+    allFrames: false,
+    includeChrome: false,
+  },
+
+  GlideHandler: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideHandlerParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideHandlerChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+        focusin: {},
+        blur: {},
+        keydown: {},
+      },
+    },
+    allFrames: true,
+    includeChrome: true,
+  },
+
   AboutLogins: {
     parent: {
       esModuleURI: "resource:///actors/AboutLoginsParent.sys.mjs",
