diff --git a/browser/components/DesktopActorRegistry.sys.mjs b/browser/components/DesktopActorRegistry.sys.mjs
index e284ae540b6222946e00173df57dfb30fba527c8..e794b450cdaee2dc6d060e524da55e8dc89d4493 100644
--- a/browser/components/DesktopActorRegistry.sys.mjs
+++ b/browser/components/DesktopActorRegistry.sys.mjs
@@ -43,6 +43,13 @@ let JSPROCESSACTORS = {
     remoteTypes: ["parent", "privilegedabout"],
   },
 
+  GeminiProtocol: {
+    parent: {
+      esModuleURI: "resource:///actors/GeminiProtocolParent.sys.mjs",
+    },
+    includeParent: true,
+  },
+
   RefreshBlockerObserver: {
     child: {
       esModuleURI: "resource:///actors/RefreshBlockerChild.sys.mjs",
@@ -96,6 +103,54 @@ let JSWINDOWACTORS = {
     remoteTypes: ["parent"],
   },
 
+  GlideDocs: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideDocsParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideDocsChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+      },
+    },
+    matches: ["resource://glide-docs/dynamic/mappings.html"],
+    allFrames: false,
+    includeChrome: false,
+  },
+  GlideTutor: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideTutorParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideTutorChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+        click: {},
+      },
+    },
+    matches: ["resource://glide-tutor/*"],
+    allFrames: false,
+    includeChrome: false,
+  },
+
+  GlideHandler: {
+    parent: {
+      esModuleURI: "resource:///actors/GlideHandlerParent.sys.mjs",
+    },
+    child: {
+      esModuleURI: "resource:///actors/GlideHandlerChild.sys.mjs",
+      events: {
+        DOMContentLoaded: {},
+        focusin: {},
+        blur: {},
+        keydown: {},
+      },
+    },
+    allFrames: true,
+    includeChrome: true,
+    safeForUntrustedWebProcess: true,
+  },
+
   AboutLogins: {
     parent: {
       esModuleURI: "resource:///actors/AboutLoginsParent.sys.mjs",
