diff --git a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
index 35f5507ea75be974d45b5adfbb5e5c0daeece38a..ed655060b0a2f07e3b7003a2e0baee627ddba6fe 100644
--- a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
+++ b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
@@ -60,6 +60,7 @@ const isMSIX =
   AppConstants.platform === "win" &&
   Services.sysinfo.getProperty("hasWinPackageId", false);
 
+// prettier-ignore
 const BASE_MESSAGES = () => [
   {
     id: "AI_WINDOW_TOU_EXISTING_USERS_MODAL",
@@ -3162,7 +3163,7 @@ const BASE_MESSAGES = () => [
     targeting:
       "('termsofuse.acceptedDate'|preferenceValue != '0') && (('termsofuse.acceptedDate'|preferenceValue * 1) < 1765972800000)",
   },
-];
+].filter((msg) => !msg.id.toLowerCase().includes('promo'));
 
 const PREONBOARDING_MESSAGES = () => [
   {
