diff --git a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
index 638d356fc26eb0e810fdafa28c1a097d3d8aee35..8b5446f597611ea40f7e0655ebe65820a4bf3633 100644
--- a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
+++ b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
@@ -3404,7 +3404,7 @@ const BASE_MESSAGES = () => [
       ],
     },
   },
-];
+].filter((msg) => !msg.id.toLowerCase().includes('promo'));
 
 const PREONBOARDING_MESSAGES = () => [
   {
