diff --git a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
index 11ddabcb1eebcfde06b086637750af829983d811..7fd684708414b7792e4d90b80a0f76683868c25f 100644
--- a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
+++ b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
@@ -60,6 +60,7 @@ const isMSIX =
   AppConstants.platform === "win" &&
   Services.sysinfo.getProperty("hasWinPackageId", false);
 
+// prettier-ignore
 const BASE_MESSAGES = () => [
   {
     id: "FXA_ACCOUNTS_BADGE",
@@ -1404,7 +1405,7 @@ const BASE_MESSAGES = () => [
     targeting:
       "(firefoxVersion >= 138 && source == 'startup' && !isDefaultBrowser && !'browser.shell.checkDefaultBrowser'|preferenceValue && currentDate|date - 'browser.shell.userDisabledDefaultCheck'|preferenceValue * 1000 >= 604800000 && isMajorUpgrade != true && platformName != 'linux' && ((currentDate|date - profileAgeCreated) / 604800000) >= 5 && !activeNotifications && 'browser.newtabpage.activity-stream.asrouter.userprefs.cfr.features'|preferenceValue && ((currentDate|date - profileAgeCreated) / 604800000) < 15",
   },
-];
+].filter((msg) => !msg.id.toLowerCase().includes('promo'));
 
 const PREONBOARDING_MESSAGES = () => [
   {
