diff --git a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
index d447d2f3858a9943996b26ef74e2feb575b790f1..9063daffa325c80010b36a090a78bfcb8cba0192 100644
--- a/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
+++ b/browser/components/asrouter/modules/OnboardingMessageProvider.sys.mjs
@@ -60,6 +60,7 @@ const isMSIX =
   AppConstants.platform === "win" &&
   Services.sysinfo.getProperty("hasWinPackageId", false);
 
+// prettier-ignore
 const BASE_MESSAGES = () => [
   {
     id: "BROWSER_BACKUP_OPTIN_SPOTLIGHT",
@@ -3035,7 +3036,7 @@ const BASE_MESSAGES = () => [
     targeting:
       "('termsofuse.acceptedDate'|preferenceValue != '0') && (('termsofuse.acceptedDate'|preferenceValue * 1) < 1765972800000)",
   },
-];
+].filter((msg) => !msg.id.toLowerCase().includes('promo'));
 
 const PREONBOARDING_MESSAGES = () => [
   {
