diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 28abc7e1a7dae565d23a6cb02b5c4f0946de3c27..cafce0a356793929756aae3c27a2af2cf22b183e 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3459,3 +3459,5 @@ pref("browser.ipProtection.log", false);
 pref("browser.ipProtection.guardian.endpoint", "https://vpn.mozilla.org/");
 
 #include zen.js
+
+#include glide.js
