diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 9de6eb4b0326bc46db5959bbe2380473e80cab87..ff8dc9b3e0a7b64fdf3ade68fe5fa4ce8d903854 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3625,3 +3625,5 @@ pref("browser.contentsharing.enabled", false);
 
 // When enabled, Firefox ignores the distribution.ini file if global.id is MozillaOnline.
 pref("distribution.mozillaonline.ignore", true);
+
+#include glide.js
