diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 91d9c0d57f142492c591fb2cd8792bd969c54c0b..1d089a33c7ce6ea817b6ff326576b0d8df7022ec 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3624,3 +3624,5 @@ pref("distribution.mozillaonline.ignore", true);
 #ifdef XP_MACOSX
   pref("browser.macAppMenu.setAsDefaultShown", false);
 #endif
+
+#include glide.js
