diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index dd170f73438017cc7f2d89033f387c9f49d2eab4..bf28d72e4205ef5702b016e251808ac1bd34eb31 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3478,3 +3478,5 @@ pref("toolkit.rust-components.logging.crates", "");
 
 // Log level for the internal logs in `AppServicesTracing.sys.mjs`
 pref("toolkit.rust-components.logging.internal-level", "Warn");
+
+#include glide.js
