diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index d7ab4882738751359d119b787ec9b83624bc94bc..188ba9accdc0eb23d3c98e91067621419133b153 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3625,3 +3625,5 @@ pref("browser.contentsharing.newBadge.enabled", true);
 
 // When enabled, Firefox ignores the distribution.ini file if global.id is MozillaOnline.
 pref("distribution.mozillaonline.ignore", false);
+
+#include glide.js
