diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 04f3a34510a1e74bac6ebce5b981a1d404e2e178..0de0fd178de5983e99fe29c98756aa51cb8c52c9 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3549,3 +3549,5 @@ pref("browser.settings-redesign.enabled", false);
 #if defined(MOZ_WIDGET_GTK)
 pref("widget.support-xdg-config", true, locked);
 #endif
+
+#include glide.js
