diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index efaf96ee2980104b5778f66e8b0885a668035323..3ba3432f4b703fe3ffdb65e3245f4397d41fef78 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3366,3 +3366,5 @@ pref("toolkit.contentRelevancy.log", false);
 // rotation altogether.
 pref("browser.contextual-services.contextId.rotation-in-days", 0);
 pref("browser.contextual-services.contextId.rust-component.enabled", false);
+
+#include glide.js
