diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index c15f009ffe137de1ab08cc2e292d0baaa00b3ae9..7e89ec5bac6cfb69543b4878612faae405f75c55 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3596,3 +3596,5 @@ pref("browser.contentsharing.enabled", false);
 
 // Controls whether the "New" badge is shown on the content sharing menu items
 pref("browser.contentsharing.newBadge.enabled", true);
+
+#include glide.js
