diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 907f272470a2da6880d722ee1dd8716ee9cc4aa4..69e1c05a9bcbf69254867bbb92c0949f074694d2 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3386,3 +3386,5 @@ pref("browser.contextual-services.contextId.rust-component.enabled", true);
 
 // Pref to enable the IP protection feature
 pref("browser.ipProtection.enabled", false);
+
+#include glide.js
