diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 920272322c92e5e075cce63ca535e99549ccb92a..4a75d69d1c31206699da4393c92debaf9c290428 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3492,3 +3492,5 @@ pref("toolkit.rust-components.logging.internal-level", "Warn");
 
 // Settings Redesign 2025 prefs
 pref("browser.settings-redesign.enabled", false);
+
+#include glide.js
