diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 53b6ab87b9b21caaecfc4fca2caa44e79ab1fdf3..eb0d62b44965f19f9d54d3747c1cb1ffa4331038 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3461,3 +3461,5 @@ pref("browser.ipProtection.enabled", false);
 pref("browser.ipProtection.variant", "");
 pref("browser.ipProtection.log", false);
 pref("browser.ipProtection.guardian.endpoint", "https://vpn.mozilla.org/");
+
+#include glide.js
