diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index beced8067e6c8b5c8cf8e3f6060ba259ec9a294d..30ee5e82c478cfa23942cd344de683c9365cdb28 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -3398,3 +3398,5 @@ pref("toolkit.contentRelevancy.enabled", false);
 pref("toolkit.contentRelevancy.ingestEnabled", false);
 // Pref to enable extra logging for the content relevancy feature
 pref("toolkit.contentRelevancy.log", false);
+
+#include glide.js
