diff --git a/browser/actors/moz.build b/browser/actors/moz.build
index 770fdca9d0affea295ce712ca7e3fd9d0138c64f..1082f807e0d1d176d964618b08216508a9d9e268 100644
--- a/browser/actors/moz.build
+++ b/browser/actors/moz.build
@@ -60,6 +60,12 @@ FINAL_TARGET_FILES.actors += [
     "EncryptedMediaParent.sys.mjs",
     "FormValidationChild.sys.mjs",
     "FormValidationParent.sys.mjs",
+    "GlideDocsChild.sys.mjs",
+    "GlideDocsParent.sys.mjs",
+    "GlideHandlerChild.sys.mjs",
+    "GlideHandlerParent.sys.mjs",
+    "GlideTutorChild.sys.mjs",
+    "GlideTutorParent.sys.mjs",
     "LightweightThemeChild.sys.mjs",
     "LinkHandlerChild.sys.mjs",
     "LinkHandlerParent.sys.mjs",
