diff --git a/build/application.ini.in b/build/application.ini.in
index 47fc729cba7d9cb612bfa0297a24644925ab53af..65e52f5431758f43d1f8a7b66d26d23e8dbed6fb 100644
--- a/build/application.ini.in
+++ b/build/application.ini.in
@@ -52,5 +52,5 @@ ServerURL=@MOZ_CRASHREPORTER_URL@/submit?id=@MOZ_APP_ID@&version=@MOZ_APP_VERSIO
 
 #if MOZ_UPDATER
 [AppUpdate]
-URL=https://@MOZ_APPUPDATE_HOST@/updates/browser/%BUILD_TARGET%/%CHANNEL%/update.xml
+URL=https://@MOZ_APPUPDATE_HOST@/updates/@MOZ_APP_VERSION@/browser/%BUILD_TARGET%/%CHANNEL%/update.xml
 #endif
