diff --git a/build/moz.build b/build/moz.build
index a84534efbc7662f81573a4a80bc045e0a6d2ed3e..228f20de31a67f8e424622ab1504e9cb9381118e 100644
--- a/build/moz.build
+++ b/build/moz.build
@@ -89,7 +89,7 @@ if CONFIG["MOZ_APP_BASENAME"]:
         if CONFIG[var]:
             appini_defines[var] = True
 
-    appini_defines["MOZ_APPUPDATE_HOST"] = "updates.zen-browser.app"
+    appini_defines["MOZ_APPUPDATE_HOST"] = "updates.glide-browser.app"
     if CONFIG["MOZ_APPUPDATE_HOST"]:
         appini_defines["MOZ_APPUPDATE_HOST"] = CONFIG["MOZ_APPUPDATE_HOST"]
 
