diff --git a/browser/components/profiles/SelectableProfileService.sys.mjs b/browser/components/profiles/SelectableProfileService.sys.mjs
index 1e62fc9395acceb57049e75db8f967d334a83ab4..9a3577fb8af833b2c0254951eb949d90aed959f9 100644
--- a/browser/components/profiles/SelectableProfileService.sys.mjs
+++ b/browser/components/profiles/SelectableProfileService.sys.mjs
@@ -227,6 +227,10 @@ class SelectableProfileServiceClass extends EventEmitter {
     this.#profileService = ProfilesDatastoreService.toolkitProfileService;
     this.#isEnabled = this.#getEnabledState();
 
+    Services.prefs.addObserver("glide.firefox.taskbar.badge.enabled", () =>
+      this.#updateTaskbar()
+    );
+
     // We have to check the state again after the policy service may have disabled us.
     Services.obs.addObserver(
       () => this.updateEnabledState(),
@@ -687,6 +691,29 @@ class SelectableProfileServiceClass extends EventEmitter {
   }
 
   async #updateTaskbar() {
+    if (
+      !Services.prefs.getBoolPref("glide.firefox.taskbar.badge.enabled", false)
+    ) {
+      if (!this.#badge) {
+        return;
+      }
+
+      // clear current badge
+      this.#badge = null;
+      if ("nsIMacDockSupport" in Ci) {
+        Cc["@mozilla.org/widget/macdocksupport;1"]
+          .getService(Ci.nsIMacDockSupport)
+          .setBadgeImage(null);
+      } else if ("nsIWinTaskbar" in Ci) {
+        for (let win of lazy.EveryWindow.readyWindows) {
+          let iconController = TASKBAR_ICON_CONTROLLERS.get(win);
+          iconController?.setOverlayIcon(null, null);
+        }
+      }
+
+      return;
+    }
+
     try {
       // We don't want the startup profile selector to badge the dock icon.
       if (!gSupportsBadging || Services.startup.startingUp) {
