diff --git a/tools/signing/macos/mach_commands.py b/tools/signing/macos/mach_commands.py
index d6b0cf119664e0534a3898f72ffbcd3aade9c89d..ac0407d07303160175066a54eb8ef1353857c3e9 100644
--- a/tools/signing/macos/mach_commands.py
+++ b/tools/signing/macos/mach_commands.py
@@ -164,6 +164,11 @@ def macos_sign(
         sys.exit(1)
     app = os.path.realpath(app_arg)
 
+    originProfile = os.path.join(os.curdir, "embedded.provisionprofile")
+    endProfile = os.path.join(app, "Contents", "embedded.provisionprofile")
+    print(f"Glide: Moving {originProfile} to {endProfile}")
+    os.rename(originProfile, endProfile)
+
     # With rcodesign, either both a p12 file and p12 password file should be
     # provided or neither. If neither, the signing identity must be either '-'
     # for ad-hoc or None.
