diff --git a/layout/generic/nsFrameSelection.cpp b/layout/generic/nsFrameSelection.cpp
index a42618dc01e15e8c8e7df4f6e5aef6f6077e77ac..547d0018d78bb34666d9bb5659d77cde99e366cf 100644
--- a/layout/generic/nsFrameSelection.cpp
+++ b/layout/generic/nsFrameSelection.cpp
@@ -517,7 +517,12 @@ nsresult nsFrameSelection::DesiredCaretPos::FetchPos(
   caret->SetSelection(&aNormalSelection);
 
   nsRect coord;
-  nsIFrame* caretFrame = caret->GetGeometry(&coord);
+  // glide-override-start
+  nsIFrame* caretFrame =
+      caret->GetGeometry(&coord, nsCaret::GeometryPositioning::ActualPosition);
+  // glide-override-original
+  // nsIFrame* caretFrame = caret->GetGeometry(&coord);
+  // glide-override-end
   if (!caretFrame) {
     return NS_ERROR_FAILURE;
   }
@@ -1862,7 +1867,12 @@ nsresult nsFrameSelection::PageMove(bool aForward, bool aExtend,
     }
 
     nsRect caretPos;
-    nsIFrame* caretFrame = nsCaret::GetGeometry(selection, &caretPos);
+    // glide-override-start
+    nsIFrame* caretFrame = nsCaret::GetGeometry(
+        selection, &caretPos, nsCaret::GeometryPositioning::Visual);
+    // glide-override-original
+    // nsIFrame* caretFrame = nsCaret::GetGeometry(selection, &caretPos);
+    // glide-override-end
     if (!caretFrame) [[unlikely]] {
       return {};
     }
