diff --git a/layout/generic/nsIFrame.cpp b/layout/generic/nsIFrame.cpp
index 44c32bd164138aeb13665aba1995ee12650d7fa5..493eb90d455e02415be35c7e1c31feba12565b07 100644
--- a/layout/generic/nsIFrame.cpp
+++ b/layout/generic/nsIFrame.cpp
@@ -11,6 +11,7 @@
 #include <stdarg.h>
 #include <algorithm>
 
+#include "mozilla/Glide.h"
 #include "gfx2DGlue.h"
 #include "gfxUtils.h"
 #include "mozilla/Attributes.h"
@@ -21,6 +22,7 @@
 #include "mozilla/EventForwards.h"
 #include "mozilla/FocusModel.h"
 #include "mozilla/StaticPrefs_dom.h"
+#include "mozilla/StaticPrefs_glide.h"
 #include "mozilla/dom/CSSAnimation.h"
 #include "mozilla/dom/CSSTransition.h"
 #include "mozilla/dom/ContentVisibilityAutoStateChangeEvent.h"
@@ -2657,7 +2659,19 @@ void nsIFrame::DisplayCaret(nsDisplayListBuilder* aBuilder,
 }
 
 nscolor nsIFrame::GetCaretColorAt(int32_t aOffset) {
-  return nsLayoutUtils::GetTextColor(this, &nsStyleUI::mCaretColor);
+  auto baseColor = nsLayoutUtils::GetTextColor(this, &nsStyleUI::mCaretColor);
+
+  // If in block caret mode, override the caret colour so that
+  // the underlying character can be visible by always using using
+  // an alpha.
+  StripAtomic<RelaxedAtomicInt32> glideCaretStyle = StaticPrefs::glide_caret_style();
+  if (StaticPrefs::glide_vimcaret() && glide::shouldRenderBlockCaret(glideCaretStyle)) {
+    auto color = sRGBColor::FromABGR(baseColor);
+    color.a = 0.5;
+    return color.ToABGR();
+  }
+
+  return baseColor;
 }
 
 auto nsIFrame::ComputeShouldPaintBackground() const -> ShouldPaintBackground {
