diff --git a/browser/base/content/aboutDialog.xhtml b/browser/base/content/aboutDialog.xhtml
index 067eaf8d4e23f76a2423d01a1d9fa71d9ef3c65f..47cfc735ccc06fc93d2ae83e8718131d2db98623 100644
--- a/browser/base/content/aboutDialog.xhtml
+++ b/browser/base/content/aboutDialog.xhtml
@@ -129,8 +129,8 @@
             <label is="text-link" useoriginprincipal="true" href="about:credits" data-l10n-name="community-creditsLink"/>
           </description>
           <description class="text-blurb" id="contributeDesc" data-l10n-id="helpus">
-            <label is="text-link" href="https://foundation.mozilla.org/?form=firefox-about" data-l10n-name="helpus-donateLink"/>
-            <label is="text-link" href="https://www.mozilla.org/contribute/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-name="helpus-getInvolvedLink"/>
+            <label is="text-link" href="https://github.com/sponsors/glide-browser" data-l10n-name="helpus-donateLink"/>
+            <label is="text-link" href="https:/github.com/glide-browser/glide" data-l10n-name="helpus-getInvolvedLink"/>
           </description>
           <description class="text-blurb" id="contributeDescReferrals" data-l10n-id="helpus-referrals" hidden="true">
             <label is="text-link" href="https://foundation.mozilla.org/?form=firefox-about" data-l10n-name="helpus-donateLink"/>
