diff --git a/browser/base/content/aboutDialog.js b/browser/base/content/aboutDialog.js
index f6e1391baf12abb91c85a95107bb3923118746c0..56593d26f05465ed9c06d117ecf54b38cd27f7ac 100644
--- a/browser/base/content/aboutDialog.js
+++ b/browser/base/content/aboutDialog.js
@@ -53,6 +53,7 @@ function init() {
   let versionIdKey = "base";
   let versionAttributes = {
     version: AppConstants.MOZ_APP_VERSION_DISPLAY,
+    firefoxVersion: AppConstants.GLIDE_FIREFOX_VERSION,
   };
 
   let arch = Services.sysinfo.get("arch");
