diff --git a/browser/base/content/browser-menubar.inc.xhtml b/browser/base/content/browser-menubar.inc.xhtml
index ea461ff47202e536021909a6ba5f6374617556c5..0d6fc2844ea3b68c24d1ea1413da56cae4261ba9 100644
--- a/browser/base/content/browser-menubar.inc.xhtml
+++ b/browser/base/content/browser-menubar.inc.xhtml
@@ -3,6 +3,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 <menubar id="main-menubar">
+  <script src="chrome://glide/content/glide-menubar.js" />
   <script src="chrome://browser/content/browser-menubar.js" />
   <menu id="file-menu" data-l10n-id="menu-file">
     <menupopup id="menu_FilePopup" accesskey-conflicts-bug="2073895">
@@ -472,6 +473,9 @@
       <menuitem id="helpPolicySupport" hidden="true"/>
     </menupopup>
   </menu>
+
+  <menu id="menu_glideMode" data-l10n-id="menu-glide-mode">
+  </menu>
 #ifdef XP_MACOSX
 <!-- nsMenuBarX hides these and uses them to build the Application menu. -->
 <!-- macOS 13 changed "Preferences" to "Settings" so one of these gets picked. -->
