diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index 40cb306b16cf6f1cb935af75a0ee20cf5ce1618a..ce0cf920659648db9573dc2ec6c6db41381e57d9 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -13,6 +13,10 @@ var { AppConstants } = ChromeUtils.importESModule(
 // lazy module getters
 
 ChromeUtils.defineESModuleGetters(this, {
+  // glide-injection-start
+  GlideHandlerParent: "resource:///actors/GlideHandlerParent.sys.mjs",
+  // glide-injection-end
+
   AIWindow:
     "moz-src:///browser/components/aiwindow/ui/modules/AIWindow.sys.mjs",
   AMTelemetry: "resource://gre/modules/AddonManager.sys.mjs",
