diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
index 0d0a559a6473149b50f4bf18d059e86907c4acd3..8ac80e8ba3cbeffeadeeb6fb2d0853e3ccfbf2be 100644
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -108,6 +108,7 @@
 # support MacOS specific features *must* go into the global-scripts.inc file so
 # that they can be shared with macWindow.inc.xhtml.
 #include global-scripts.inc
+#include ../../../glide/browser/base/content/glide-scripts.inc
 
 <script>
   /* eslint-env mozilla/browser-window */
