diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
index dd2f391096968564086911c532f696dc7ffb8ae1..9957640fd9d08798303a1b9dcf07ca2d1663f57d 100644
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -148,6 +148,18 @@
                        command="Browser:Stop"
                        tooltip="dynamic-shortcut-tooltip"/>
       </toolbaritem>
+
+      <toolbarbutton id="glide-toolbar-mode-button" class="toolbarbutton-1 chromeclass-toolbar-additional glide-mode-button"
+                     badged="true"
+                     delegatesanchor="true"
+                     consumeanchor="glide-toolbar-mode-button"
+                     closemenu="none"
+                     data-l10n-id="glide-toolbar-button-mode"
+                     cui-areatype="toolbar"
+                     removable="true">
+        normal
+      </toolbarbutton>
+
       <toolbarspring cui-areatype="toolbar" class="chromeclass-toolbar-additional"/>
       <toolbarspring cui-areatype="toolbar"
                    class="chromeclass-toolbar-additional titlebar-spacer"
@@ -453,6 +465,15 @@
         </box>
       </toolbarbutton>
 
+      <toolbarbutton id="glide-toolbar-keyseq-button" class="toolbarbutton-1 chromeclass-toolbar-additional glide-keyseq-button"
+                     badged="true"
+                     delegatesanchor="true"
+                     consumeanchor="glide-toolbar-keyseq-button"
+                     closemenu="none"
+                     data-l10n-id="glide-toolbar-button-keyseq"
+                     cui-areatype="toolbar"
+                     removable="true">
+      </toolbarbutton>
       <toolbarbutton id="fxa-toolbar-menu-button" class="toolbarbutton-1 chromeclass-toolbar-additional subviewbutton-nav"
                      badged="true"
                      delegatesanchor="true"
