diff --git a/browser/base/content/navigator-toolbox.js b/browser/base/content/navigator-toolbox.js
index c6fefefef50c73bcfb2e4b2ac347741d3f3d5cbc..fbca8f2f93400703e592e974066c720c8e644b08 100644
--- a/browser/base/content/navigator-toolbox.js
+++ b/browser/base/content/navigator-toolbox.js
@@ -11,6 +11,23 @@ document.addEventListener(
     const navigatorToolbox = document.getElementById("navigator-toolbox");
     const widgetOverflow = document.getElementById("widget-overflow");
 
+    // glide-injection-start
+
+    // The default CSS adds a child node for the icon but as we don't have an icon
+    // it doesn't work. So just remove it here as the alternative is defining our own CSS class which
+    // will likely come with a lot of maintenance burden and this appears to be the only thing that doesn't
+    // work out of the box at the time of writing (2024-11-15).
+    const glide_mode_button = document.getElementById(
+      "glide-toolbar-mode-button"
+    );
+    glide_mode_button.removeChild(glide_mode_button.childNodes[1]);
+    const glide_keyseq_button = document.getElementById(
+      "glide-toolbar-keyseq-button"
+    );
+    glide_keyseq_button.removeChild(glide_keyseq_button.childNodes[0]);
+
+    // glide-injection-end
+
     function onPopupShowing(event) {
       switch (event.target.id) {
         case "PlacesChevronPopup":
