diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
index 30e7b9b2ac63db6ccd2727a9341081cecefc25cb..50310b0d787d7e930e32672314d0425cebaa3871 100644
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -156,6 +156,22 @@
           </box>
         </toolbarbutton>
       </toolbaritem>
+
+      <!-- glide-injection-start -->
+      <!-- TODO(glide): should this be a button / popover menu? -->
+      <!-- TODO(glide): should this be removable? -->
+      <toolbarbutton id="glide-toolbar-mode-button" class="toolbarbutton-1 chromeclass-toolbar-additional glide-mode-button"
+                     badged="true"
+                     delegatesanchor="true"
+                     consumeanchor="glide-toolbar-mode-button"
+                     closemenu="none"
+                     data-l10n-id="glide-toolbar-button-mode"
+                     cui-areatype="toolbar"
+                     removable="true">
+        normal
+      </toolbarbutton>
+      <!-- glide-injection-end -->
+
       <toolbarspring cui-areatype="toolbar" class="chromeclass-toolbar-additional"/>
       <toolbarspring cui-areatype="toolbar" class="chromeclass-toolbar-additional titlebar-spacer" id="vertical-spacer" />
       <toolbaritem id="urlbar-container"
@@ -455,6 +471,15 @@
         </box>
       </toolbarbutton>
 
+      <toolbarbutton id="glide-toolbar-keyseq-button" class="toolbarbutton-1 chromeclass-toolbar-additional glide-keyseq-button"
+                     badged="true"
+                     delegatesanchor="true"
+                     consumeanchor="glide-toolbar-keyseq-button"
+                     closemenu="none"
+                     data-l10n-id="glide-toolbar-button-keyseq"
+                     cui-areatype="toolbar"
+                     removable="true">
+      </toolbarbutton>
       <toolbarbutton id="fxa-toolbar-menu-button" class="toolbarbutton-1 chromeclass-toolbar-additional subviewbutton-nav"
                      badged="true"
                      delegatesanchor="true"
