diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
index 754181405c8f44c672a8c78eda341e422be5e62b..5e4528fa551ab4da00fe300b80356ae656c287d5 100644
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -155,6 +155,18 @@
           </box>
         </toolbarbutton>
       </toolbaritem>
+
+      <toolbarbutton id="glide-toolbar-mode-button" class="toolbarbutton-1 chromeclass-toolbar-additional glide-mode-button"
+                     badged="true"
+                     delegatesanchor="true"
+                     consumeanchor="glide-toolbar-mode-button"
+                     closemenu="none"
+                     data-l10n-id="glide-toolbar-button-mode"
+                     cui-areatype="toolbar"
+                     removable="true">
+        normal
+      </toolbarbutton>
+      
       <toolbarspring cui-areatype="toolbar" class="chromeclass-toolbar-additional"/>
       <toolbarspring cui-areatype="toolbar"
                    class="chromeclass-toolbar-additional titlebar-spacer"
@@ -450,6 +462,15 @@
         </box>
       </toolbarbutton>
 
+      <toolbarbutton id="glide-toolbar-keyseq-button" class="toolbarbutton-1 chromeclass-toolbar-additional glide-keyseq-button"
+                     badged="true"
+                     delegatesanchor="true"
+                     consumeanchor="glide-toolbar-keyseq-button"
+                     closemenu="none"
+                     data-l10n-id="glide-toolbar-button-keyseq"
+                     cui-areatype="toolbar"
+                     removable="true">
+      </toolbarbutton>
       <toolbarbutton id="fxa-toolbar-menu-button" class="toolbarbutton-1 chromeclass-toolbar-additional subviewbutton-nav"
                      badged="true"
                      delegatesanchor="true"
