diff --git a/toolkit/mozapps/extensions/AddonManager.sys.mjs b/toolkit/mozapps/extensions/AddonManager.sys.mjs
index 4fe6aaa127d12bfebb9b246cf1bcaeb78aeef506..0ed265fbebd20938611cd07d8a89ffaebe742a45 100644
--- a/toolkit/mozapps/extensions/AddonManager.sys.mjs
+++ b/toolkit/mozapps/extensions/AddonManager.sys.mjs
@@ -1279,6 +1279,9 @@ var AddonManagerInternal = {
         ITEM_STATUS: addonStatus,
         APP_ID: Services.appinfo.ID,
         APP_VERSION: aAppVersion ? aAppVersion : Services.appinfo.version,
+        FIREFOX_VERSION: aAppVersion
+          ? aAppVersion
+          : AppConstants.GLIDE_FIREFOX_VERSION,
         REQ_VERSION: UPDATE_REQUEST_VERSION,
         APP_OS: Services.appinfo.OS,
         APP_ABI: Services.appinfo.XPCOMABI,
@@ -1780,6 +1783,7 @@ var AddonManagerInternal = {
    * @param  {object} [aOptions.telemetryInfo]
    *         An optional object which provides details about the installation source
    *         included in the addon manager telemetry events.
+   * @returns {Promise<AddonInstall>}
    * @throws if aUrl is not specified or if an optional argument of
    *         an improper type is passed.
    */
@@ -4368,6 +4372,7 @@ export var AddonManager = {
     return "unknown";
   },
 
+  /** @type {AddonManagerInternal['getInstallForURL']} */
   getInstallForURL(aUrl, aOptions) {
     return AddonManagerInternal.getInstallForURL(aUrl, aOptions);
   },
