diff --git a/toolkit/mozapps/extensions/AddonManager.sys.mjs b/toolkit/mozapps/extensions/AddonManager.sys.mjs
index 2c1e187d14bae0dcc17a1fa2de768e7c949b3776..12e73215918f2869a6bcbacb356e4429fc3fbf4a 100644
--- a/toolkit/mozapps/extensions/AddonManager.sys.mjs
+++ b/toolkit/mozapps/extensions/AddonManager.sys.mjs
@@ -1228,6 +1228,9 @@ var AddonManagerInternal = {
         ITEM_STATUS: addonStatus,
         APP_ID: Services.appinfo.ID,
         APP_VERSION: aAppVersion ? aAppVersion : Services.appinfo.version,
+        FIREFOX_VERSION: aAppVersion
+          ? aAppVersion
+          : AppConstants.GLIDE_FIREFOX_VERSION,
         REQ_VERSION: UPDATE_REQUEST_VERSION,
         APP_OS: Services.appinfo.OS,
         APP_ABI: Services.appinfo.XPCOMABI,
@@ -1728,6 +1731,7 @@ var AddonManagerInternal = {
    * @param  {Object} [aOptions.telemetryInfo]
    *         An optional object which provides details about the installation source
    *         included in the addon manager telemetry events.
+   * @returns {Promise<AddonInstall>}
    * @throws if aUrl is not specified or if an optional argument of
    *         an improper type is passed.
    */
@@ -4315,6 +4319,7 @@ export var AddonManager = {
     return "unknown";
   },
 
+  /** @type {AddonManagerInternal['getInstallForURL']} */
   getInstallForURL(aUrl, aOptions) {
     return AddonManagerInternal.getInstallForURL(aUrl, aOptions);
   },
