diff --git a/mobile/ios/moz.configure b/mobile/ios/moz.configure
index 035ab10e38c71d30d3404209f1404be7d7351c67..91b3a8c910205acafeca4a4e4696b585147d480f 100644
--- a/mobile/ios/moz.configure
+++ b/mobile/ios/moz.configure
@@ -6,7 +6,11 @@
 
 imply_option("MOZ_GECKOVIEW_HISTORY", True)
 
-imply_option("MOZ_APP_VENDOR", "Mozilla")
+# glide-override-start
+imply_option("MOZ_APP_VENDOR", "Glide")
+# glide-original
+# imply_option("MOZ_APP_VENDOR", "Mozilla")
+# glide-override-end
 imply_option("MOZ_APP_ID", "{8e3ba928-3798-4bcd-9b64-c788f2d954d3}")
 
 
