diff --git a/toolkit/themes/shared/desktop-jar.inc.mn b/toolkit/themes/shared/desktop-jar.inc.mn
index 530d95c9c32f8c056946c85c22335bb546cbedf3..49faf5936f9267aa27afe1f9d86fb32cf6d2b883 100644
--- a/toolkit/themes/shared/desktop-jar.inc.mn
+++ b/toolkit/themes/shared/desktop-jar.inc.mn
@@ -34,6 +34,9 @@
   skin/classic/global/design-system/tokens-shared.css      (../../shared/design-system/tokens-shared.css)
   skin/classic/global/error-pages.css                      (../../shared/error-pages.css)
   skin/classic/global/findbar.css                          (../../shared/findbar.css)
+  skin/classic/global/glide.css                            (../../shared/glide.css)
+  skin/classic/global/glide-hints.css                      (../../shared/glide-hints.css)
+  skin/classic/global/glide-commandline.css                (../../shared/glide-commandline.css)
   skin/classic/global/global-shared.css                    (../../shared/global-shared.css)
   skin/classic/global/menu.css                             (../../shared/menu.css)
   skin/classic/global/menulist-shared.css                  (../../shared/menulist-shared.css)
