diff --git a/toolkit/themes/shared/desktop-jar.inc.mn b/toolkit/themes/shared/desktop-jar.inc.mn
index 9f41a9206d65664a0ff481d874e80a7b02a2f25d..e12e0ce2c1b5022b45abe61bec3892981c340d59 100644
--- a/toolkit/themes/shared/desktop-jar.inc.mn
+++ b/toolkit/themes/shared/desktop-jar.inc.mn
@@ -34,6 +34,9 @@
   skin/classic/global/design-system/tokens-shared.css      (../../shared/design-system/tokens-shared.css)
   skin/classic/global/error-pages.css                      (../../shared/error-pages.css)
   skin/classic/global/findbar.css                          (../../shared/findbar.css)
+  skin/classic/global/glide.css                            (../../shared/glide.css)
+  skin/classic/global/glide-hints.css                      (../../shared/glide-hints.css)
+  skin/classic/global/glide-commandline.css                (../../shared/glide-commandline.css)
   skin/classic/global/global-shared.css                    (../../shared/global-shared.css)
   skin/classic/global/menu.css                             (../../shared/menu.css)
   skin/classic/global/menulist.css                         (../../shared/menulist.css)
