diff --git a/toolkit/themes/shared/desktop-jar.inc.mn b/toolkit/themes/shared/desktop-jar.inc.mn
index 8857ca35818d82886bed547e771d955d41a8ce06..c5eb33376bb9de1d3e9f05eb22fc7bf0a030bebd 100644
--- a/toolkit/themes/shared/desktop-jar.inc.mn
+++ b/toolkit/themes/shared/desktop-jar.inc.mn
@@ -33,6 +33,9 @@
   skin/classic/global/design-system/tokens-shared.css      (../../shared/design-system/dist/tokens-shared.css)
   skin/classic/global/error-pages.css                      (../../shared/error-pages.css)
   skin/classic/global/findbar.css                          (../../shared/findbar.css)
+  skin/classic/global/glide.css                            (../../shared/glide.css)
+  skin/classic/global/glide-hints.css                      (../../shared/glide-hints.css)
+  skin/classic/global/glide-commandline.css                (../../shared/glide-commandline.css)
   skin/classic/global/global-shared.css                    (../../shared/global-shared.css)
   skin/classic/global/menu.css                             (../../shared/menu.css)
   skin/classic/global/menulist.css                         (../../shared/menulist.css)
