diff --git a/tools/lint/eslint/eslint-plugin-mozilla/lib/rules/reject-import-system-module-from-non-system.mjs b/tools/lint/eslint/eslint-plugin-mozilla/lib/rules/reject-import-system-module-from-non-system.mjs
index 1519915676e3899dd77ff9825e76bb685c7b1421..d2c39b3e6ffc9b9d0240d5caeba4325a0ea2cce2 100644
--- a/tools/lint/eslint/eslint-plugin-mozilla/lib/rules/reject-import-system-module-from-non-system.mjs
+++ b/tools/lint/eslint/eslint-plugin-mozilla/lib/rules/reject-import-system-module-from-non-system.mjs
@@ -24,6 +24,11 @@ export default {
           return;
         }
 
+        if (node.importKind === "type") {
+          // importing types doesn't have any runtime semantics, which is what this rule is for.
+          return;
+        }
+
         context.report({
           node,
           messageId: "rejectStaticImportSystemModuleFromNonSystem",
