diff --git a/toolkit/modules/UpdateUtils.sys.mjs b/toolkit/modules/UpdateUtils.sys.mjs
index 2c8a093013d40623aa16864eca18945887add7c8..25da46442805801a505fdc0bd164b292fb83211d 100644
--- a/toolkit/modules/UpdateUtils.sys.mjs
+++ b/toolkit/modules/UpdateUtils.sys.mjs
@@ -83,6 +83,9 @@ export var UpdateUtils = {
         case "VERSION":
           replacement = Services.appinfo.version;
           break;
+        case "FIREFOX_VERSION":
+          replacement = AppConstants.GLIDE_FIREFOX_VERSION;
+          break;
         case "BUILD_ID":
           replacement = Services.appinfo.appBuildID;
           break;
@@ -98,6 +101,10 @@ export var UpdateUtils = {
         case "CHANNEL":
           replacement = this.UpdateChannel;
           break;
+        case "FIREFOX_CHANNEL":
+          // TODO(glide): extract this to a constant
+          replacement = "beta";
+          break;
         case "PLATFORM_VERSION":
           replacement = Services.appinfo.platformVersion;
           break;
