diff --git a/browser/components/extensions/parent/ext-sidebarAction.js b/browser/components/extensions/parent/ext-sidebarAction.js
index 3b6cf76d226bd9db07b829d33118dac71545fb99..dd850a022b9c1062b9eaf63311a45d20c27cb13a 100644
--- a/browser/components/extensions/parent/ext-sidebarAction.js
+++ b/browser/components/extensions/parent/ext-sidebarAction.js
@@ -88,7 +88,7 @@ this.sidebarAction = class extends ExtensionAPI {
       // sidebar_action not specified for this extension.
       return;
     }
-    this.tabContext.shutdown();
+    this.tabContext?.shutdown();
     // Don't remove everything on app shutdown so session restore can handle
     // restoring open sidebars.
     if (isAppShutdown) {
