diff --git a/tools/@types/generated/lib.gecko.dom.d.ts b/tools/@types/generated/lib.gecko.dom.d.ts
index 7361392796bec5caa11de4592b8a6212682aa017..8fadd9d011666e804520b95f8d874d944bec1ffa 100644
--- a/tools/@types/generated/lib.gecko.dom.d.ts
+++ b/tools/@types/generated/lib.gecko.dom.d.ts
@@ -24682,6 +24682,7 @@ declare var XULCommandEvent: {
 };
 
 interface XULElementEventMap extends ElementEventMap, GlobalEventHandlersEventMap, OnErrorEventHandlerForNodesEventMap, TouchEventHandlersEventMap {
+    keypress: KeyboardEvent
 }
 
 interface XULElement extends Element, ElementCSSInlineStyle, ElementOffsetAttributes, GlobalEventHandlers, HTMLOrForeignElement, OnErrorEventHandlerForNodes, TouchEventHandlers {
