diff --git a/tools/@types/generated/lib.gecko.xpcom.d.ts b/tools/@types/generated/lib.gecko.xpcom.d.ts
index dfade9b260bdb4f5f7ac2c05cafa4714e580d865..8701df4221ac71baaca1203d6cae37120a3dc0e7 100644
--- a/tools/@types/generated/lib.gecko.xpcom.d.ts
+++ b/tools/@types/generated/lib.gecko.xpcom.d.ts
@@ -2626,6 +2626,7 @@ interface nsIDOMWindowUtils extends nsISupports, Enums<typeof nsIDOMWindowUtils_
   getScrollXYFloat(aFlushLayout: boolean, aScrollX: OutParam<float>, aScrollY: OutParam<float>): void;
   getScrollbarSize(aFlushLayout: boolean, aWidth: OutParam<i32>, aHeight: OutParam<i32>): void;
   getBoundsWithoutFlushing(aElement: Element): DOMRect;
+  getElementBoundingScreenRect(aElement: Element): DOMRect;
   getWidgetOpaqueRegion(): DOMRect[];
   scrollToVisual(aOffsetX: float, aOffsetY: float, aUpdateType: i32, aScrollMode: i32): void;
   getVisualViewportOffsetRelativeToLayoutViewport(aOffsetX: OutParam<float>, aOffsetY: OutParam<float>): void;
@@ -7415,7 +7416,7 @@ type nsISTSShutdownObserver = Callable<{
 }>
 
 interface nsISocketTransportService extends nsISupports {
-  createTransport(aSocketTypes: string[], aHost: string, aPort: i32, aProxyInfo: nsIProxyInfo, dnsRecord: nsIDNSRecord): nsISocketTransport;
+  createTransport(aSocketTypes: string[], aHost: string, aPort: i32, aProxyInfo: nsIProxyInfo | null, dnsRecord: nsIDNSRecord | null): nsISocketTransport;
   createUnixDomainTransport(aPath: nsIFile): nsISocketTransport;
   createUnixDomainAbstractAddressTransport(aName: string): nsISocketTransport;
 }
