diff --git a/tools/@types/generated/lib.gecko.dom.d.ts b/tools/@types/generated/lib.gecko.dom.d.ts
index 025d03e5a4942efd603cb4e5fd93a722db57900c..4be6b92d68c2294c82e6574a72fe8d3887cefeec 100644
--- a/tools/@types/generated/lib.gecko.dom.d.ts
+++ b/tools/@types/generated/lib.gecko.dom.d.ts
@@ -29065,31 +29065,6 @@ declare var ReadableByteStreamController: {
     isInstance: IsInstance<ReadableByteStreamController>;
 };
 
-/** <!-- binding_to(idl, class, WEBIDL_ReadableStream) --> */
-interface ReadableStream {
-    /** <!-- binding_to(idl, attribute, WEBIDL_ReadableStream_locked) --> */
-    readonly locked: boolean;
-    /** <!-- binding_to(idl, method, WEBIDL_ReadableStream_cancel) --> */
-    cancel(reason?: any): Promise<void>;
-    /** <!-- binding_to(idl, method, WEBIDL_ReadableStream_getReader) --> */
-    getReader(options?: ReadableStreamGetReaderOptions): ReadableStreamReader;
-    /** <!-- binding_to(idl, method, WEBIDL_ReadableStream_pipeThrough) --> */
-    pipeThrough(transform: ReadableWritablePair, options?: StreamPipeOptions): ReadableStream;
-    /** <!-- binding_to(idl, method, WEBIDL_ReadableStream_pipeTo) --> */
-    pipeTo(destination: WritableStream, options?: StreamPipeOptions): Promise<void>;
-    /** <!-- binding_to(idl, method, WEBIDL_ReadableStream_tee) --> */
-    tee(): ReadableStream[];
-}
-
-declare var ReadableStream: {
-    prototype: ReadableStream;
-    new(underlyingSource?: any, strategy?: QueuingStrategy): ReadableStream;
-    /** <!-- binding_to(idl, attribute, WEBIDL_ReadableStream_isInstance) --> */
-    isInstance: IsInstance<ReadableStream>;
-    /** <!-- binding_to(idl, method, WEBIDL_ReadableStream_from) --> */
-    from(asyncIterable: any): ReadableStream;
-};
-
 /** <!-- binding_to(idl, class, WEBIDL_ReadableStreamBYOBReader) --> */
 interface ReadableStreamBYOBReader extends ReadableStreamGenericReader {
     /** <!-- binding_to(idl, method, WEBIDL_ReadableStreamBYOBReader_read) --> */
@@ -41940,6 +41915,7 @@ declare var XULCommandEvent: {
 };
 
 interface XULElementEventMap extends ElementEventMap, GlobalEventHandlersEventMap, OnErrorEventHandlerForNodesEventMap, TouchEventHandlersEventMap {
+    keypress: KeyboardEvent
 }
 
 /** <!-- binding_to(idl, class, WEBIDL_XULElement) --> */
@@ -45421,11 +45397,6 @@ interface FileSystemDirectoryHandle {
     values(): AsyncIterableIterator<FileSystemHandle>;
 }
 
-interface ReadableStream {
-    [Symbol.asyncIterator](options?: ReadableStreamIteratorOptions): AsyncIterableIterator<any>;
-    values(options?: ReadableStreamIteratorOptions): AsyncIterableIterator<any>;
-}
-
 interface TestInterfaceAsyncIterableDouble {
     [Symbol.asyncIterator](): AsyncIterableIterator<[string, string]>;
     entries(): AsyncIterableIterator<[string, string]>;
