diff --git a/tools/@types/generated/lib.gecko.xpcom.d.ts b/tools/@types/generated/lib.gecko.xpcom.d.ts
index a783ef95fc219734642ffe0331c4de714faa7675..f901462daa54f8543ec3233f0808b14a8c268ad3 100644
--- a/tools/@types/generated/lib.gecko.xpcom.d.ts
+++ b/tools/@types/generated/lib.gecko.xpcom.d.ts
@@ -12741,7 +12741,7 @@ type nsISTSShutdownObserver = Callable<{
 /** <!-- binding_to(idl, interface_name, XPIDL_nsISocketTransportService) --> */
 interface nsISocketTransportService extends nsISupports {
   /** <!-- binding_to(idl, method, XPIDL_nsISocketTransportService_createTransport) --> */
-  createTransport(aSocketTypes: string[], aHost: string, aPort: i32, aProxyInfo: nsIProxyInfo, dnsRecord: nsIDNSRecord): nsISocketTransport;
+  createTransport(aSocketTypes: string[], aHost: string, aPort: i32, aProxyInfo: nsIProxyInfo | null, dnsRecord: nsIDNSRecord | null): nsISocketTransport;
   /** <!-- binding_to(idl, method, XPIDL_nsISocketTransportService_createUnixDomainTransport) --> */
   createUnixDomainTransport(aPath: nsIFile): nsISocketTransport;
   /** <!-- binding_to(idl, method, XPIDL_nsISocketTransportService_createUnixDomainAbstractAddressTransport) --> */
