diff --git a/modules/libpref/init/all.js b/modules/libpref/init/all.js
index e0b2092b7a9271cdf3ed863a0f5b991bb19cfe76..d2dc024b0faa4cb7654aab3ece514d030c47b150 100644
--- a/modules/libpref/init/all.js
+++ b/modules/libpref/init/all.js
@@ -3484,7 +3484,7 @@ pref("browser.search.removeEngineInfobar.enabled", true);
 pref("media.gmp-manager.allowLocalSources", true);
 
 // Update service URL for GMP install/updates:
-pref("media.gmp-manager.url", "https://aus5.mozilla.org/update/3/GMP/%VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/update.xml");
+pref("media.gmp-manager.url", "https://aus5.mozilla.org/update/3/GMP/%PLATFORM_VERSION%/%BUILD_ID%/%BUILD_TARGET%/%LOCALE%/%FIREFOX_CHANNEL%/%OS_VERSION%/%DISTRIBUTION%/%DISTRIBUTION_VERSION%/update.xml");
 
 // Whether or not to perform reader mode article parsing on page load.
 // If this pref is disabled, we will never show a reader mode icon in the toolbar.
@@ -4050,6 +4050,15 @@ pref("toolkit.osKeyStore.loglevel", "Warn");
 
 pref("extensions.formautofill.supportRTL", false);
 
+// ----- glide-injection-start -----
+
+// Glide loggers
+
+pref("glide.logging.loglevel", "Error");
+pref("glide.gemini.loglevel", "Error");
+
+// ----- glide-injection-end -----
+
 // Controls the log level for CookieBannerListService.sys.mjs.
 pref("cookiebanners.listService.logLevel", "Error");
 
