diff --git a/mobile/android/moz.configure b/mobile/android/moz.configure
index dad9a16c7ab21f3a31ce034fd1bb4d31cc58a8ae..4ad718f22f74052d584cfc969ff97e6fde1a6f21 100644
--- a/mobile/android/moz.configure
+++ b/mobile/android/moz.configure
@@ -82,7 +82,11 @@ imply_option("MOZ_SERVICES_HEALTHREPORT", True)
 imply_option("MOZ_GECKOVIEW_HISTORY", True)
 
 imply_option("MOZ_APP_UA_NAME", "Firefox")
-imply_option("MOZ_APP_VENDOR", "Mozilla")
+# glide-override-start
+imply_option("MOZ_APP_VENDOR", "Glide")
+# glide-original
+# imply_option("MOZ_APP_VENDOR", "Mozilla")
+# glide-override-end
 imply_option("MOZ_APP_ID", "{aa3c5121-dab2-40e2-81ca-7ea25febc110}")
 imply_option("BROWSER_CHROME_URL", "chrome://geckoview/content/geckoview.xhtml")
 
