diff --git a/browser/components/customizableui/CustomizableUI.sys.mjs b/browser/components/customizableui/CustomizableUI.sys.mjs
index 4aca0aa0c90603e09b58395debb62875c018a8cf..8a20aeead8cf4447a694f6c06c473e475bcb8bc5 100644
--- a/browser/components/customizableui/CustomizableUI.sys.mjs
+++ b/browser/components/customizableui/CustomizableUI.sys.mjs
@@ -358,6 +358,7 @@ var CustomizableUIInternal = {
       "back-button",
       "forward-button",
       "stop-reload-button",
+      "glide-toolbar-mode-button",
       Services.policies.isAllowed("removeHomeButtonByDefault")
         ? null
         : "home-button",
@@ -366,6 +367,7 @@ var CustomizableUIInternal = {
       "urlbar-container",
       "spring",
       "downloads-button",
+      "glide-toolbar-keyseq-button",
       AppConstants.MOZ_DEV_EDITION ? "developer-button" : null,
       lazy.ippEnabled ? "ipprotection-button" : null,
       "fxa-toolbar-menu-button",
