diff --git a/browser/components/customizableui/CustomizableUI.sys.mjs b/browser/components/customizableui/CustomizableUI.sys.mjs
index 132c77e396cb259181ed13ca8ff784e0ade05e3b..9ab700f2e2d2f0bc8a4d2687fc801a7e60a9677e 100644
--- a/browser/components/customizableui/CustomizableUI.sys.mjs
+++ b/browser/components/customizableui/CustomizableUI.sys.mjs
@@ -334,10 +334,12 @@ var CustomizableUIInternal = {
       "back-button",
       "forward-button",
       "stop-reload-button",
+      "glide-toolbar-mode-button",
       "spring",
       "vertical-spacer",
       "urlbar-container",
       "spring",
+      "glide-toolbar-keyseq-button",
       AppConstants.MOZ_DEV_EDITION ? "developer-button" : null,
       lazy.resetPBMToolbarButtonEnabled ? "reset-pbm-toolbar-button" : null,
     ].filter(name => name);
