diff --git a/browser/components/customizableui/CustomizableUI.sys.mjs b/browser/components/customizableui/CustomizableUI.sys.mjs
index 5bb6be19f7f855b129aac921af1ed5cfc63d732b..0b0b958fa9594c0611236a9472981e5534cb8514 100644
--- a/browser/components/customizableui/CustomizableUI.sys.mjs
+++ b/browser/components/customizableui/CustomizableUI.sys.mjs
@@ -331,6 +331,7 @@ var CustomizableUIInternal = {
       "back-button",
       "forward-button",
       "stop-reload-button",
+      "glide-toolbar-mode-button",
       Services.policies.isAllowed("removeHomeButtonByDefault")
         ? null
         : "home-button",
@@ -340,6 +341,7 @@ var CustomizableUIInternal = {
       "spring",
       "save-to-pocket-button",
       "downloads-button",
+      "glide-toolbar-keyseq-button",
       AppConstants.MOZ_DEV_EDITION ? "developer-button" : null,
       "fxa-toolbar-menu-button",
       lazy.resetPBMToolbarButtonEnabled ? "reset-pbm-toolbar-button" : null,
