diff --git a/netwerk/base/NetUtil.sys.mjs b/netwerk/base/NetUtil.sys.mjs
index 555deac6e7e399875b24eb87c6156c1d4b88526f..538e83574ffa08bb755234684147b8c3f8831d4b 100644
--- a/netwerk/base/NetUtil.sys.mjs
+++ b/netwerk/base/NetUtil.sys.mjs
@@ -83,7 +83,7 @@ export var NetUtil = {
    * is asynchronous, I/O may happen on the main thread.  When reading from
    * a local file, prefer using IOUtils methods instead.
    *
-   * @param {nsIChannel|nsIInputStream|NewChannelOptions} aSource
+   * @param {any} aSource
    *        This argument can be one of the following:
    *         - An options object that will be passed to NetUtil.newChannel.
    *         - An existing nsIChannel.
