diff --git a/browser/locales/en-US/browser/browserContext.ftl b/browser/locales/en-US/browser/browserContext.ftl
index 51ed054ea5b75d86da9c329842bcd2eb285a0424..7af1f3067e48a061e2866408ed1147e9116af262 100644
--- a/browser/locales/en-US/browser/browserContext.ftl
+++ b/browser/locales/en-US/browser/browserContext.ftl
@@ -81,6 +81,27 @@ toolbar-button-stop =
 toolbar-button-stop-reload =
     .title = { main-context-menu-reload.aria-label }
 
+########### glide-injection-start
+
+glide-toolbar-button-mode =
+    .label = Glide current mode
+    .tooltiptext = Glide current mode
+
+glide-toolbar-button-keyseq =
+    .label = Glide current key sequence
+    .tooltiptext = Glide current key sequence
+
+glide-error-notification-reload-config-button =
+    .label = :config_reload
+
+glide-notification-copy-to-clipboard-button =
+    .label = Copy to clipboard
+
+glide-error-notification-clear-all-button =
+    .label = :clear
+
+########### glide-injection-end
+
 ## Account toolbar Button
 
 toolbar-button-account =
