diff --git a/browser/locales/en-US/browser/browserContext.ftl b/browser/locales/en-US/browser/browserContext.ftl
index 4b95b09620bee676b308d684e11efbd99a9ac6b0..79ac75b3b9e338bfbb33c10c7899bc3dd4f22936 100644
--- a/browser/locales/en-US/browser/browserContext.ftl
+++ b/browser/locales/en-US/browser/browserContext.ftl
@@ -81,6 +81,17 @@ toolbar-button-stop =
 toolbar-button-stop-reload =
     .title = { main-context-menu-reload.aria-label }
 
+########### glide-injection-start
+
+glide-toolbar-button-mode =
+    .label = Glide current mode
+    .tooltiptext = Glide current mode
+
+glide-error-notification-reload-config-button =
+    .label = :config_reload
+
+########### glide-injection-end
+
 ## Account toolbar Button
 
 toolbar-button-account =
