diff --git a/browser/locales/en-US/browser/browserContext.ftl b/browser/locales/en-US/browser/browserContext.ftl
index 4b95b09620bee676b308d684e11efbd99a9ac6b0..6f211d545e296403e9b6cd4cc9419e53b215a0cb 100644
--- a/browser/locales/en-US/browser/browserContext.ftl
+++ b/browser/locales/en-US/browser/browserContext.ftl
@@ -81,6 +81,23 @@ toolbar-button-stop =
 toolbar-button-stop-reload =
     .title = { main-context-menu-reload.aria-label }
 
+########### glide-injection-start
+
+glide-toolbar-button-mode =
+    .label = Glide current mode
+    .tooltiptext = Glide current mode
+
+glide-error-notification-reload-config-button =
+    .label = :config_reload
+
+glide-config-notification-copy-button =
+    .label = Copy to clipboard
+
+glide-error-notification-clear-all-button =
+    .label = Clear all
+
+########### glide-injection-end
+
 ## Account toolbar Button
 
 toolbar-button-account =
